pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -287,9 +287,9 @@ func (r *Reconciler) fetchAllResourcesInOneNamespace(namespaceName string, place
 	return resources, nil
 }
 
-// shouldSelectResource returns whether a resource should be propagated
+// shouldSelectResource returns whether a resource should be selected for propagation.
 func (r *Reconciler) shouldSelectResource(gvr schema.GroupVersionResource) bool {
-	if r.DisabledResourceConfig == nil {
+	if r.ResourceConfig.IsEmpty() {
```

Comment:
`What if the isAllow is true and resourceConfig is empty? we should not select the resource.` - Agreed.

However the ResourceConfig will never be a nil, instead an empty struct is returned by the following function, right? Hence I added another method to check if the ResourceConfig is actually empty of any resources in it. 
```
func NewResourceConfig(isAllowList bool) *ResourceConfig {
	r := &ResourceConfig{
		groups:            map[string]struct{}{},
		groupVersions:     map[schema.GroupVersion]struct{}{},
		groupVersionKinds: map[schema.GroupVersionKind]struct{}{},
	}
```

Reply:
To make it clear, there are two issues.

1. about "nil", right now, we always initialize the reconciler using NewResourceConfig(), but you cannot prevent other caller creating a reconciler without setting the resourceConfig. It's better to add the check to prevent the regression/panic in the future.
2. about adding r.ResourceConfig.IsEmpty() as a shortcut, i'm fine with checking whether it's empty or not as a shortcut though it's redundant and existing code should cover that.

If you want to add that, it should be

1) expose the isAllowList value
2) and then based on this value to return,

eg, if r.ResourceConfig.IsEmpty() {
      return !r.ResourceConfig.isAllowList
}

Could you please add a UT to test this func to prevent the regression? today we have the tests covered for the SkippedPropagatingAPIs list, but not for the AllowedPropagatingAPIs.

Reply:
Thanks for the explanation! I have kept the original nil check since `r.ResourceConfig.IsResourceDisabled(gvk) should return correctly if the list is empty.`